src/deroff.rs

```diff
@@ -1060,23 +1164,56 @@ fn test_comment() {
     assert_eq!(deroffer.s, "\nworld".to_owned());
 }
 
-fn deroff_files(files: &[String]) -> std::io::Result<()> {
+fn deroff_files<P: AsRef<std::path::Path>, S: ToString>(
+    files: &[S],
+    output_dir: P,
+) -> std::io::Result<()> {
+    let output_path = output_dir.as_ref().to_path_buf();
     for arg in files {
+        let arg = arg.to_string();
         eprintln!("processing deroff file: {}", arg);
 
-        let mut file = File::open(arg)?;
+        let mut file = File::open(&arg)?;
         let mut string = String::new();
         if arg.ends_with(".gz") {
             let mut decoder = GzDecoder::new(file);
             decoder.read_to_string(&mut string)?;
         } else {
-            file.read_to_string(&mut string)?;
+            match file.read_to_string(&mut string) {
+                Err(_) => {
+                    // TODO: This is a _bad_ workaround for latin1, we need to correctly decode input files
+                    let mut bytes = Vec::new();
+                    file.read_to_end(&mut bytes)?;
+
+                    for byte in bytes {
+                        string.push(byte as char);
+                    }
+                }
+                _ => {}
+            };
```

Comment:
Is this even necessary? I wonder if we could drop this feature altogether, who uses `latin1` nowdays? UTF-8 for the win.

Reply:
The original explicitly supports python, I think we should outdo them by supporting more encodings. I'm thinking about another module that will take a byte stream in any encoding and convert it to a utf-8 string, thoughts?

Reply:
I dont know why it sent two different messages there but basically
the python supports latin1, we should do more, maybe a thing in util or something (post-mvp)

Reply:
> Is this even necessary? I wonder if we could drop this feature altogether, who uses `latin1` nowdays? UTF-8 for the win.

Regarding `latin1` support, we have a couple samples in rust-dc/manpages that use `latin1`. For MVP, wouldn't want to strip functionality compared to the original unless the original functionality is inherently buggy or insecure.

Reply:
https://github.com/rust-dc/manpages is latin only, I wonder what will happen if we add non-latin manpages into the mix. We throw everything from my /usr/share/man there. I think I have like 30609 man pages on my machine, sometimes I do try to read chinese manpages for fun. We could also use it for benchmark later on other than just fuzzing.

Reply:
> https://github.com/rust-dc/manpages is latin only

Ah, yeah, I meant that there are only a few examples (that I knew of, at least) that were using characters in the Latin-1 non-ASCII range. Everything else is only trivially `latin1`.

> I wonder what will happen if we add non-latin manpages into the mix. We throw everything from my /usr/share/man there. I think I have like 30609 man pages on my machine, sometimes I do try to read chinese manpages for fun. We could also use it for benchmark later on other than just fuzzing.

For sure. It would make sense to expand to other encodings.